### ISSUE TITLE
Over indentation when autocorrecting `Layout/FirstHashElementLineBreak`

### ISSUE THREAD

**Issue by sambostock** [rubocop/rubocop#11462]
In certain circumstances, when autocorrecting `Layout/FirstHashElementLineBreak`, the resulting `Hash` is way over indented.

I have found this tricky to minimally reproduce, as it seems to involve multiple cops. I was able to distill it down to this:

```yaml
# .rubocop.yml
AllCops:
  SuggestExtensions: false
  DisabledByDefault: true
Style/FrozenStringLiteralComment:
  Enabled: false

Layout/FirstHashElementIndentation:
  Enabled: true
Layout/FirstHashElementLineBreak:
  Enabled: true
Layout/TrailingWhitespace:
  Enabled: true
```

```ruby
# example.rb
def example
  { key1:
    { key2:
      { key3:
        { key4:
          { key5:
            :value
          }
        }
      }
    }
  }
end
```

--------

## Expected behavior

Running `rubocop -a` should produce an output with reasonable indentation. Something like

```ruby
# example.rb
def example
  {
    key1:
      {
        key2:
          {
            key3:
              {
                key4:
                  {
                    key5:
                      :value
                  }
              }
          }
      }
  }
end
```

A human would probably manually correct it as this , but I'm not sure if they're a setting to move the `{`  onto the previous line.

```ruby
# example.rb
def example
  {
    key1: {
      key2: {
        key3: {
          key4: {
            key5: :value
          }
        }
      }
    }
  }
end
```

In either scenario, the indentation is reasonable (2 spaces indented compared to parent line).

## Actual behavior

Running `rubocop -a` over indents the contents of the hash:

```ruby
# example.rb
def example
  {
    key1:
        {
          key2:
                {
                  key3:
                          {
                            key4:
                                      {
                                        key5:
                                                    :value
                                      }
                          }
                }
        }
  }
end
```

Notice the indentation seems compound and get worse the more it is nested. It appears
- the value for `:key1` is over nested by 2 spaces relative to the `key1:` line
- the value for `:key2` is over nested by 4 spaces relative to the `key2:` line
- the value for `:key3` is over nested by 6 spaces relative to the `key3:` line
- the value for `:key4` is over nested by 8 spaces relative to the `key4:` line
- the value for `:key5` is over nested by 10 spaces relative to the `key5:` line

<details><summary><code>rubocop --debug -a</code> output</summary>

```
bundle exec rubocop --debug -a
For /Users/sambostock/src/local/hash-indentation-rubocop: configuration from /Users/sambostock/src/local/hash-indentation-rubocop/.rubocop.yml
Default configuration from /Users/sambostock/.gem/ruby/3.2.0/gems/rubocop-1.43.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/sambostock/.cache/rubocop_cache/b430b43db81d34665407c9c16a480bb2aa4cbd18/6d7a3b621ca1730e04accd938619e4bdab66cfb1/deb763edf0b8905d2918766c77e22932380bb7d2
Inspecting 2 files
Scanning /Users/sambostock/src/local/hash-indentation-rubocop/Gemfile
Loading cache from /Users/sambostock/.cache/rubocop_cache/b430b43db81d34665407c9c16a480bb2aa4cbd18/6d7a3b621ca1730e04accd938619e4bdab66cfb1/deb763edf0b8905d2918766c77e22932380bb7d2
.Scanning /Users/sambostock/src/local/hash-indentation-rubocop/example.rb
Loading cache from /Users/sambostock/.cache/rubocop_cache/b430b43db81d34665407c9c16a480bb2aa4cbd18/6d7a3b621ca1730e04accd938619e4bdab66cfb1/279d36ebd03288310c8ecbbd18929a90c667d984
C

Offenses:

example.rb:3:4: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
  { 
   ^
example.rb:3:5: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
  { key1: ...
    ^^^^^
example.rb:4:1: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
key1: ...
^^^^^
example.rb:4:7: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
    { key2: ...
      ^^^^^
example.rb:5:6: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
    { 
     ^
example.rb:5:9: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
      { key3: ...
        ^^^^^
example.rb:6:1: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
key2: ...
^^^^^
example.rb:6:11: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
        { key4: ...
          ^^^^^
example.rb:7:8: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
      { 
       ^
example.rb:7:13: C: [Corrected] Layout/FirstHashElementLineBreak: Add a line break before the first element of a multi-line hash.
          { key5: ...
            ^^^^^
example.rb:8:1: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
key3: ...
^^^^^
example.rb:9:10: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
        { 
         ^
example.rb:10:1: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
key4: ...
^^^^^
example.rb:11:12: C: [Corrected] Layout/TrailingWhitespace: Trailing whitespace detected.
          { 
           ^
example.rb:12:1: C: [Corrected] Layout/FirstHashElementIndentation: Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.
key5: ...
^^^^^

2 files inspected, 15 offenses detected, 15 offenses corrected
Finished in 1.838947999291122 seconds
```

</details>

## Steps to reproduce the problem

See `.rubocop.yml` and `example.rb` above.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.43.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin21.5.0]
```

---

Originally reported by @vytis in Shopify/ruby-style-guide#482.

**Comment by sambostock** [comment]
I tried to write a spec for this, but I couldn't figure out how to get multiple cops to run in a single spec. Is there somewhere where we do "integration tests" to see how different cops work together to correct things?

**Comment by dvandersluis** [comment]
Yes: https://github.com/rubocop/rubocop/blob/master/spec/rubocop/cli/autocorrect_spec.rb

**Comment by sambostock** [comment]
I've added an integration test in #11463. I'm not familiar with the internals of the autocorrector, so if anyone else is, assistance would be appreciated! 🙏
